src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -0,0 +1,242 @@
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public abstract class BaseTermQueryBuilder<QB extends BoostableQueryBuilder<QB>> extends BaseQueryBuilder implements Streamable, BoostableQueryBuilder<QB> {
+    
+    /** Name of field to match against. */
+    protected String fieldName;
+
+    /** Value to find matches for. */
+    protected Object value;
+
+    /** Query boost. */
+    protected float boost = 1.0f;
+
+    /** Name of the query. */
+    protected String queryName;
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, String value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, int value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, long value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, float value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, double value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, boolean value) {
+        this(fieldName, (Object) value);
+    }
+
+    /**
+     * Constructs a new base term query.
+     *
+     * @param fieldName  The name of the field
+     * @param value The value of the term
+     */
+    public BaseTermQueryBuilder(String fieldName, Object value) {
+        this.fieldName = fieldName;
+        if (value instanceof String) {
+            this.value = BytesRefs.toBytesRef(value);
+        } else {
+            this.value = value;
+        }
+    }
+
+    BaseTermQueryBuilder() {
+        // for serialization only
+    }
+
+    /** Returns the field name used in this query @see org.apache.lucene.search.spans.SpanTermQuery#getField() */
```

Comment:
Not sure I used the correct syntax. My goal was to provide a link to the original Lucene documentation. Do you know if there is a way to reference JavaDoc of another project?

Doesn't need to be a real hyperlink but it would be nice if (power?) users could get a hint which parameter this ends up being in the resulting Lucene query.